### ISSUE TITLE
validation loss

### ISSUE THREAD

**Issue by SherrySwift** [fvmassoli/cross-resolution-face-recognition#3]
Hi, thx for the excellent job! 
I try to run the code but I encounter a problem. The validation loss becomes very large from the begining of the training, and the validation accuracy is always zero.
The following information is part of my training.log:

############
2020-09-22 23:13:52,822 | VGGFace2 custom training dataset info:
				Root folder:       ./VGG-Face2/data/train
				Downsampling prob: 0.1
				Use Curriculum:    True
				Valid resolution:  8
2020-09-22 23:13:53,930 | VGGFace2 custom validation dataset info:
				Root folder:       ./VGG-Face2/data/validation
				Downsampling prob: 1.0
				Use Curriculum:    False
				Valid resolution:  8
2020-09-22 23:13:55,161 | Train datasets initialized!!!
2020-09-22 23:13:55,161 | Initializing VGGFace2 data loaders...
2020-09-22 23:13:55,163 | VGGFace2 data summary:
2020-09-22 23:13:55,163 | 	Batch size:                  32
				Number of workers:           8
				Training images:             3141890
				Training batches:            98185
				Validation images:           2824
				Validation batches:          89
				Validation original images:  2824
				Validation original batches: 89
				Pin Memory:                  False

2020-09-22 23:14:07,917 | Valid loss HR: 15.437 --- Valid acc HR: 0.00%
2020-09-22 23:14:21,509 | Valid loss LR: 14.460 --- Valid acc LR: 0.00%
2020-09-22 23:14:21,513 | ##############################
2020-09-22 23:14:21,513 | Training at epoch: 1
2020-09-22 23:14:21,513 | ##############################
2020-09-22 23:14:27,106 | Train [1] - [8]/[98185]:
			Loss LR: 2.887 --- Acc LR: 69.92%
			curr_index: 1 --- downsampling_prob: 0.10000000149011612
2020-09-22 23:14:29,841 | Train [1] - [16]/[98185]:
			Loss LR: 4.308 --- Acc LR: 51.56%
			curr_index: 33 --- downsampling_prob: 0.10000000149011612
2020-09-22 23:14:42,295 | Valid loss HR: 19.352 --- Valid acc HR: 0.00%
2020-09-22 23:14:56,317 | Valid loss LR: 11.447 --- Valid acc LR: 0.00%
2020-09-22 23:14:57,970 | Train [1] - [24]/[98185]:
			Loss LR: 4.944 --- Acc LR: 44.27%
			curr_index: 65 --- downsampling_prob: 0.10000000149011612
2020-09-22 23:14:59,732 | Train [1] - [32]/[98185]:
			Loss LR: 5.351 --- Acc LR: 38.77%
			curr_index: 97 --- downsampling_prob: 0.10000000149011612
2020-09-22 23:15:11,963 | Valid loss HR: 218881.020 --- Valid acc HR: 0.00%
2020-09-22 23:15:25,458 | Valid loss LR: 104311.859 --- Valid acc LR: 0.00%
2020-09-22 23:15:27,102 | Train [1] - [40]/[98185]:
			Loss LR: 5.514 --- Acc LR: 35.62%
			curr_index: 129 --- downsampling_prob: 0.10000000149011612
2020-09-22 23:15:28,872 | Train [1] - [48]/[98185]:
			Loss LR: 5.581 --- Acc LR: 34.77%
			curr_index: 161 --- downsampling_prob: 0.10000000149011612
2020-09-22 23:15:41,319 | Valid loss HR: 11224037.615 --- Valid acc HR: 0.00%
2020-09-22 23:15:55,085 | Valid loss LR: 7921513.108 --- Valid acc LR: 0.00%
############

Valid loss becomes larger as the training procedure goes on. 
What could be reason for this? 
Thx a lot.

**Comment by fvmassoli** [comment]
Hi,
I just tried to run the code and it works perfectly. See below

2020-09-23 13:00:26,642 | Initializing VGGFace2 data sets...
2020-09-23 13:00:31,900 | VGGFace2 custom training dataset info:                                                                                  
                                Root folder:       ./
                                Downsampling prob: 0.1
                                Use Curriculum:    False
                                Valid resolution:  8
2020-09-23 13:00:32,564 | VGGFace2 custom validation dataset info:                                                                                
                                Root folder:      ./
                                Downsampling prob: 1.0
                                Use Curriculum:    False
                                Valid resolution:  8
2020-09-23 13:00:33,313 | Train datasets initialized!!!
2020-09-23 13:00:33,313 | Initializing VGGFace2 data loaders...
2020-09-23 13:00:33,313 | VGGFace2 data summary:
2020-09-23 13:00:33,313 |       Batch size:                  32
                                Number of workers:           8
                                Training images:             2831606
                                Training batches:            88488
                                Validation images:           5172
                                Validation batches:          162
                                Validation original images:  5172
                                Validation original batches: 162
                                Pin Memory:                  False

2020-09-23 13:00:34,528 | Valid loss HR: 0.003 --- Valid acc HR: 93.75%                                                                           
2020-09-23 13:00:35,418 | Valid loss LR: 0.100 --- Valid acc LR: 0.00%                                                                            
2020-09-23 13:00:35,422 | ##############################
2020-09-23 13:00:35,422 | Training at epoch: 1
2020-09-23 13:00:35,422 | ##############################
2020-09-23 13:00:37,439 | Train [1] - [1]/[88488]:
                        Loss LR: 2.116 --- Acc LR: 71.88%
                        curr_index: 0 --- downsampling_prob: 0.10000000149011612
2020-09-23 13:00:38,176 | Valid loss HR: 0.003 --- Valid acc HR: 90.62%                                                                           
2020-09-23 13:00:39,100 | Valid loss LR: 0.070 --- Valid acc LR: 0.00%                                                                            
2020-09-23 13:00:39,220 | Train [1] - [2]/[88488]:
                        Loss LR: 1.310 --- Acc LR: 85.94%
                        curr_index: 0 --- downsampling_prob: 0.10000000149011612
2020-09-23 13:00:39,964 | Valid loss HR: 0.005 --- Valid acc HR: 84.38%                                                                           
2020-09-23 13:00:40,891 | Valid loss LR: 0.063 --- Valid acc LR: 0.00%                                                                            
2020-09-23 13:00:41,012 | Train [1] - [3]/[88488]:
                        Loss LR: 1.341 --- Acc LR: 85.42%
                        curr_index: 0 --- downsampling_prob: 0.10000000149011612
2020-09-23 13:00:41,763 | Valid loss HR: 0.007 --- Valid acc HR: 81.25%                                                                           
2020-09-23 13:00:42,722 | Valid loss LR: 0.063 --- Valid acc LR: 0.00%                                                                            
2020-09-23 13:00:42,842 | Train [1] - [4]/[88488]:
                        Loss LR: 1.365 --- Acc LR: 85.16%
                        curr_index: 0 --- downsampling_prob: 0.10000000149011612
2020-09-23 13:00:43,580 | Valid loss HR: 0.010 --- Valid acc HR: 71.88%                                                                           
2020-09-23 13:00:44,501 | Valid loss LR: 0.064 --- Valid acc LR: 0.00% 


Could it be that you used a too high learning rate?

As a double check, run the code with learning rate = 0 and see if the problem disappear

**Comment by SherrySwift** [comment]
Hi, 
Thanks for your suggestion. I change the learning rate from 0.001 to 0.0001, now the validation loss becomes more reasonable than before.
But I still have a problem. the value of valid loss is almost unchanged during training (Valid loss HR is around 16, Valid loss LR is around 10), and the valid accuracies still remains 0s. 
So I guess there might be something wrong with the dataset I use. Could you please tell me how you make the validation dataset? I just download the 'test' dataset from the official website of VGGFace2 and use it as 'validation' set in this experiment. I don't know if it is right to do so.

**Comment by fvmassoli** [comment]
The VGGFace2 test contains 500 identities that are not present into the training set since it is meant to test the model on the face recognition protocols and not on the classification task.
1. Take a small portion of the training set as validation set
2. You should never use the test set while training a model

**Comment by SherrySwift** [comment]
Thanks a lot! I'll reorganize the dataset and try again.
